### PR TITLE
Questions about your refactoring

### PR DESCRIPTION
**Pull request by JeremyRms**

### CODE REVIEW

File: ApiController.php [hunk 1]
```diff
@@ -16,12 +16,14 @@
...
     {
 	// What kind of problem can we have when using global variables?
+//Concurrency issues//Testing and Confinement 
         $Case = $_GET["case"];
         $customer = $_POST["customer"];
         $product_id = $_GET["productId"];
 
         try {
 	// think about what could be done to avoid this logic in a controller
+	// try to move to services layer
```
Comment (JeremyRms) [inline ApiController.php]: Exact, but how?

File: ApiController.php [hunk 2]
```diff
@@ -16,12 +16,14 @@
     public function doAction(CustomerReposiotry customer)
     {
 	// What kind of problem can we have when using global variables?
+//Concurrency issues//Testing and Confinement
```
Comment (JeremyRms) [inline ApiController.php]: Concurrency issues? Can you explain?

File: ApiController.php [hunk 3]
```diff
@@ -3,7 +3,7 @@
     private $dataBase = null;
 
     // What would be a correct way to write this constructor?
-    public ApiController(IDatabase $dataBase,LoggerInterface $logger) {
+    public function ApiController(IDatabase $dataBase,LoggerInterface $logger) {
```
Comment (JeremyRms) [inline ApiController.php]: It's not how you create a constructor in PHP

Comment (nguyenjay90) [reply ApiController.php]: ok  fix now .try to change .
function __construct (IDatabase $dataBase,LoggerInterface $logger)
{
///if ($database == null)
            throw new DatabaseException ("not could not connect to database error ");
//you can look at IDatabase 
        $this->dataBase = $dataBase;
		$this->logger=$logger
}

File: ApiController.php (hunk 2 above)
Comment (nguyenjay90) [reply ApiController.php]: Yes Concurrency issues .If using  globals variable can be access by multiple threads of execution, synchronization is necessary. When dynamically linking modules with globals, the composed system might not be thread-safe even if the two independent modules tested in dozens of different contexts were safe.

File: ApiController.php (hunk 1 above)
Comment (nguyenjay90) [reply ApiController.php]: in this case. i think we can apply factory method design patern and move to services layer .. not have logic in controller.

Comment (JeremyRms) [reply ApiController.php]: That's a way, yes. Can you try to use it here?

Comment (nguyenjay90) [reply ApiController.php]: class Factory
{
    public function CreateProcess($type)
    {
        if ($type == 'product_id')
		{
            retrun new Product();
        } 
		if ($type == ''contact'')
		{
            retrun new Contact();
        } else{
            return new customer();
        }
    }
}
Factory method 
abstract Class MyProcess
{
    abstract function CreateProcess();

    public function Process()
    {
        $Product = $this->Create();
        ...
        ...
    }
}
///

class ProductProcess extends MyProcess
{
    function Create()
    {
        return new product();
    }
}
///
....

Comment (JeremyRms) [reply ApiController.php]: Push a pull request please, in the git-flow way if possible

Comment (nguyenjay90) [reply ApiController.php]: i want to remind you .this is comment.

Comment (JeremyRms) [reply ApiController.php]: What do you mean?

Comment (nguyenjay90) [reply ApiController.php]: i want to explain for you  .how to implement Factory method it here

Comment (JeremyRms) [reply ApiController.php]: Yes, but can you actually code it?

Comment (nguyenjay90) [reply ApiController.php]: you can review again .this is actully code. .

**Pull request merged**